core/src/io/anuke/mindustry/game/Rules.java

```diff
@@ -45,4 +50,12 @@
     public byte zone = -1;
     /** Spawn layout. Since only zones modify this, it should be assigned on save load. */
     public transient Array<SpawnGroup> spawns = DefaultWaves.get();
+    /** Determines if there should be limited respawns. */
+    public boolean limitedRespawns = false;
+    /** How many times player can respawn during one wave. */
+    public int respawns = 5;
+    /** Hold wave timer until all enemies are destroyed. */
+    public boolean waitForWaveToEnd = false;
+    /** Determinates if gamemode requires more than 1 cores to play */
+    public boolean manyCores = false;
```

Comment:
I don't think this variable name makes sense. What it really controls is the *objective* of the map, not whether it requires multiple cores to play. Something like `attackMode` might be better, but even then I'm not too happy with hardcoded objectives like this. A flexible `Objective` enum would be the best solution, but that's outside the scope of this PR.

Reply:
`attackMode` is a perfect solution for now. `Objective` enum would be great, but also `Gamemode` is kind of `Objective` itself. How I mentioned below, that's why `Gamemode` is no longer only `RulePreset `

Reply:
Following this way, we can also delete PVP rule, and handle logic by checking game's gamemode instead of single rule. Maybe that's the way.

Reply:
But this kind of rework would be useful only if we want to introduce new mixed modes. For now, I guess it's sufficient.